src/lib/TableSizeParser.ts

```diff
@@ -0,0 +1,78 @@
+const stringWidth = require('string-width');
+
+export default class TableSizeParser {
+	padString(str : string, length : number) {
+		let a = (length / 2) - (str.length / 2);
+		return str.padStart(a + str.length).padEnd(length);
+	}
+
+	stringLength(str : string) {
+		// todo: get correct width of emojis
+		return stringWidth(str);
+	}
+
+	tableHeading(people) {
+		let longestReceiver = this.stringLength('receiver');
+		let longestSender = this.stringLength('sender');
+		let longestComment = this.stringLength('comments');
+		people.forEach((person) => {
+			if (this.stringLength(person.person) > longestReceiver) {
+				longestReceiver = this.stringLength(person.person);
+			}
+			person.pegs.forEach((pegger) => {
+				if (this.stringLength(pegger.sender) > longestSender) {
+					longestSender = this.stringLength(pegger.sender);
+				}
+				if (this.stringLength(pegger.comment) > longestComment) {
+					longestComment = this.stringLength(pegger.comment);
+				}
+			});
+		});
+
+		// define table heading
+		let tableHeading = this.padString('Receiver', longestReceiver) + ' | ' + this.padString('Sender', longestSender) + ' | Comments\n';
+		tableHeading += 'Total'.padEnd(longestReceiver) + ' | ' + ' '.padEnd(longestSender) + ' | \n';
+		tableHeading += ''.padEnd(longestReceiver, '-') + '-+-' + ''.padEnd(longestSender, '-') + '-+-' + ''.padEnd(longestComment, '-') + '\n';
+		return {
+			heading: tableHeading,
+			longestReceiver,
+			longestSender,
+			longestComment
+		};
+	}
+}
+
+exports.padString = function(str : string, length : number) {
+	let a = (length / 2) - (str.length / 2);
+	return str.padStart(a + str.length).padEnd(length);
+}
+
+exports.stringLength = function(str : string) {
+	// todo: get correct width of emojis
+	return stringWidth(str);
+}
+
+exports.tableHeading = function(people) {
```

Comment:
Oh wait, this file is entirely just the same three functions exported as part of a class and separately...

Reply:
I'm struggling to find places where these functions have been used at all actually. 'results.ts' and 'winners.ts' appear to just take in the TableSizeParser in their constructors and not actually use it.